lib/Sema/CSDiagnostics.cpp

```diff
@@ -7292,6 +7293,8 @@ bool InOutConversionFailure::diagnoseAsError() {
       assert(locator->findLast<LocatorPathElt::ContextualType>());
       auto anchor = getAnchor();
       auto contextualType = getContextualType(anchor);
+      if (contextualType.isNull())
```

Comment:
This indicates that the fix was anchored on an incorrect position if there is no contextual type recorded for a `ContextualType` element, we'd need to address that during solving and make this check an assertion instead.